tests/components/homekit/test_homekit.py

```diff
@@ -1051,3 +1052,75 @@ def _mock_get_accessory(*args, **kwargs):
             "linked_motion_sensor": "binary_sensor.camera_motion_sensor",
         },
     )
+
+
+async def test_homekit_finds_linked_humidity_sensors(
+    hass, hk_driver, debounce_patcher, device_reg, entity_reg
+):
+    """Test HomeKit start method."""
+    entry = await async_init_integration(hass)
+
+    homekit = HomeKit(
+        hass,
+        None,
+        None,
+        None,
+        {},
+        {"humidifier.humidifier": {}},
+        DEFAULT_SAFE_MODE,
+        advertise_ip=None,
+        entry_id=entry.entry_id,
+    )
+    homekit.driver = hk_driver
+    homekit._filter = Mock(return_value=True)
+    homekit.bridge = HomeBridge(hass, hk_driver, "mock_bridge")
+
+    config_entry = MockConfigEntry(domain="test", data={})
+    config_entry.add_to_hass(hass)
+    device_entry = device_reg.async_get_or_create(
+        config_entry_id=config_entry.entry_id,
+        sw_version="0.16.1",
+        model="Smart Brainy Clever Humidifier",
+        manufacturer="Home Assistant",
+        connections={(device_registry.CONNECTION_NETWORK_MAC, "12:34:56:AB:CD:EF")},
+    )
+
+    humidity_sensor = entity_reg.async_get_or_create(
+        "sensor",
+        "humidifier",
+        "humidity_sensor",
+        device_id=device_entry.id,
+        device_class=DEVICE_CLASS_HUMIDITY,
+    )
+    humidifier = entity_reg.async_get_or_create(
+        "humidifier", "humidifier", "demo", device_id=device_entry.id
+    )
+
+    hass.states.async_set(
+        humidity_sensor.entity_id, "42%", {ATTR_DEVICE_CLASS: DEVICE_CLASS_HUMIDITY},
```

Comment:
```suggestion
        humidity_sensor.entity_id, 42, {ATTR_DEVICE_CLASS: DEVICE_CLASS_HUMIDITY},
```

And then set ATTR_UNIT_OF_MEASUREMENT as well